enhancements/network/dpu-ovnkube-operator.md

```diff
@@ -0,0 +1,202 @@
+---
+title: DPU OVNKube Operator
+authors:
+  - "@pliurh"
+reviewers:
+  - "@zshi-redhat"
+  - "@dcbw"
+approvers:
+  - 
+creation-date: 2021-09-07
+last-updated: 2021-09-07
+status: implementable
+---
+
+# DPU OVNKube Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+To facilitate the management of Nvidia BlueField-2 DPU, a two-cluster design is
+proposed. Under such design, a BlueFiled-2 card will be provisioned as a worker
+node of the ARM-based infra cluster, whereas the tenant cluster where the normal
+user applications run on, is composed of the X86 servers.
+
+The OVN-Kubernetes components are spread over the two clusters. On the tenant
+cluster side, the Cluster Network Operator is in charge of the management of the
+ovn-kube components. On the infra cluster side, we propose to create a new
+operator to be responsible for the life-cycle management of ovn-kube components
+and the necessary host network initialization on DPU cards.
+
+## Motivation
+
+Normally, the Cluster Network Operator watches the Network customer resource,
+then provision the ovn-kube components to the Openshift cluster.
```

Comment:
Normally CNO manages the ovn-kube components. If this new operator is taking over that role for the infra cluster, do we need to document somewhere that CNO needs to be changed to not manage OVN-Kube on infra clusters and what's the CNO trigger for that?

Reply:
This operator is not taking take over that role for the infra cluster, but managing the ovnkube that servers the tenant cluster. The CNO will still manage the ovnkube of the infra cluster on non-BF2 nodes.

@zshi-redhat Shall we mention these CNO changes in https://github.com/openshift/enhancements/pull/732

Reply:
I think we need a similar patch in CNO like this https://github.com/openshift/cluster-network-operator/pull/1138, to isolate the CNO managed ovnkube from being scheduled on DPU in infra cluster. For example, to introduce the label `network.operator.openshift.io/dpu` which will be used by CNO in infra cluster.

Reply:
@pliurh yes, THE #732 needs to be updated.